stream-chat-swift-docs/docs/guides/working-with-user.md

```diff
@@ -1,3 +1,134 @@
+```swift 
+userController.mute()
+userController.unmute()
+```
+Both function have optional completion block to be called when the network request is finished
+```swift 
+userController.mute {
+    if let error = error {
+        print(error)
+    }
+    // Successfully finished
+}
+```
+
+### Flagging/Unflagging user
+```swift 
+userController.flag()
+userController.unflag()
+```
+Both function have optional completion block to be called when the network request is finished
+```swift 
+userController.flag {
+    if let error = error {
+        print(error)
+    }
+    // Successfully finished
+}
+```
+
+## Updating current user data
+
+You can call update user data function on current user controller
+
+```swift 
+currentUserController.updateUserData(name: "Luke Skywalker")
+```
+
+You can observe the changes in user data with delegate.
+
+```swift 
+// Don't forget to keep reference to the delegate. 
+// `CurrentUserController` keeps only weak reference to the delegate.
+let delegate = YourDelegate()
+currentUserController.delegate = delegate
+
+currentUserController.updateUserData(name: "Luke Skywalker")
+
+class YourDelegate: CurrentChatUserControllerDelegate {
+    func currentUserController(
+        _ controller: _CurrentChatUserController<ExtraData>,
+        didChangeCurrentUser: EntityChange<_CurrentChatUser<ExtraData>>
+    ) {
+        if case let .update(user) = didChangeCurrentUser {
+            print(user.name) // Luke Skywalker
+        }
+    }
+}
+```
+You can use combine to observe the changes too
+```swift
+currentUserController
+    .currentUserChangePublisher
+    .compactMap({ (change: EntityChange<CurrentChatUser>) -> CurrentChatUser? in
+        if case let .update(user) = change { return user } else { return nil }
+    })
+    .sink { print($0.name) } // Luke Skywalker
+    .store(in: &cancellables)
+```
+
+## Observe unread count for user
+
+To observe unread count on current user controller, you need to set a delegate. 
+
+```swift
+class YourDelegate: CurrentChatUserControllerDelegate {
+    func currentUserController(
+        _ controller: CurrentChatUserController, 
+        didChangeCurrentUserUnreadCount count: UnreadCount
+    ) {
+        /// Handle the undread count
+        UIApplication.shared.applicationIconBadgeNumber = count.messages
+    }
+}
+
+// Don't forget to keep reference to the delegate. 
+// `CurrentUserController` keeps only weak reference to the delegate.
+let delegate = YourDelegate()
+currentUserController.delegate = delegate
+```
+
+You can use combine to observe unread count too
+```swift
+currentUserController
+    .unreadCountPublisher
+    .map(\.messages)
+    .sink { UIApplication.shared.applicationIconBadgeNumber = $0 }
+    .store(in: &cancellables)
+```
+
+## Change user
+
+To change the logged in user, simply change token provider of the chatClient.
+
+```swift 
+let token = ... // Obtain the chat token from your system
+chatClient.tokenProvider = .static(token)
+```
```

Comment:
To change the current user we also have to initiate the reload via current user controller:
```swift
// Reload the current user
chatClient.currentUserController().reloadUserIfNeeded { error in
 ...
}
```

I'd also add some lines about `ChatClientConfig.shouldConnectAutomatically` flag, how it works, and how to manually control when client `connects/disconnects` from web-socket but that's the advanced topic 🤔 

@b-onc WDYT? Are we gonna have a dedicated page for `TokenProvider`, `reloadUserIfNeeded`, `connect/disconnect` stuff or we can briefly describe this in `current user` guide?

Reply:
@evsaev good points, we'll have a dedicated Config page so i wouldn't put the flag here. `TokenProvider` and the rest belongs to Getting Started guide

Reply:
`reloadUserIfNeeded` definitely belongs in this guide, but not the rest